src/function/scalar/string/like.cpp

```diff
@@ -1,40 +1,58 @@
-#include "duckdb/function/scalar/string_functions.hpp"
-
 #include "duckdb/common/exception.hpp"
 #include "duckdb/common/vector_operations/vector_operations.hpp"
+#include "duckdb/function/scalar/string_functions.hpp"
 
 using namespace std;
 
 namespace duckdb {
 
 static bool like_operator(const char *s, const char *pattern, const char *escape);
 
+struct LikeEscapeOperator {
+	template <class TA, class TB, class TC> static inline bool Operation(TA str, TB pattern, TC escape) {
+		// Only one escape character should be allowed
+		if (escape.GetSize() > 1) {
+			throw SyntaxException("Invalid escape string. Escape string must be empty or one character.");
+		}
+		return like_operator(str.GetData(), pattern.GetData(), escape.GetData());
+	}
+};
+
+struct NotLikeEscapeOperator {
+	template <class TA, class TB, class TC> static inline bool Operation(TA str, TB pattern, TC escape) {
+		return !LikeEscapeOperator::Operation(str, pattern, escape);
+	}
+};
+
 struct LikeOperator {
-	template <class TA, class TB, class TR> static inline TR Operation(TA left, TB right) {
-		return like_operator(left.GetData(), right.GetData(), nullptr);
+	template <class TA, class TB, class TR> static inline TR Operation(TA str, TB pattern) {
+		return like_operator(str.GetData(), pattern.GetData(), nullptr);
 	}
 };
 
 struct NotLikeOperator {
-	template <class TA, class TB, class TR> static inline TR Operation(TA left, TB right) {
-		return !like_operator(left.GetData(), right.GetData(), nullptr);
+	template <class TA, class TB, class TR> static inline TR Operation(TA str, TB pattern) {
+		return !like_operator(str.GetData(), pattern.GetData(), nullptr);
+		;
 	}
 };
 
 bool like_operator(const char *s, const char *pattern, const char *escape) {
```

Comment:
Can you revert the entire like_operator function to how it was before? The is_escaped variable is not necessary I think.